src/Components/DynamicContent/PromotionBuilder/PromotionToolBar.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropsTypes from 'prop-types'
+import isEqual from 'lodash.isequal'
+
+import { arrayToMap } from '../../../utils/utils'
+
+
+const PromotionToolBar = (props) => {
+
+    const promotionData = promotionData;
+    const checkAllFieldHasData = formData => { return !(formData.devices.size===0 || formData.ventures.size===0 || formData.url==="" || formData.name==="")}
```

Comment:
I would discourage the functions being defined like this. This would enforce re-definition of the functions every time this component renders. Would recommend creating pure functions which can be extracted to another file.